stograde/student/markdownify/process_file.py

```diff
@@ -55,10 +55,11 @@ def compile_file(filename, *, steps, results, supporting_dir, basedir, spec_id):
     return True
 
 
-def test_file(filename, *, spec, results, options, cwd, supporting_dir, interact):
-    tests = flatten([test_spec['commands']
-                     for test_spec in spec.get('tests', {})
-                     if test_spec['filename'] == filename])
+def test_file(filename, *, spec, tests, results, options, cwd, supporting_dir, interact):
+    tests = tests if spec.get('spec_version', 2) >= 3 else \
+        flatten([test_spec['commands']
+                 for test_spec in spec.get('tests', {})
+                 if test_spec['filename'] == filename])
```

Comment:
Let's split this into a few lines…

```suggestion
	if spec.get('spec_version', 2) >= 3:
		tests = tests
	else:
    	tests = flatten([test_spec['commands']
                 		 for test_spec in spec.get('tests', {})
                 		 if test_spec['filename'] == filename])
```

Reply:
I figured out a better way that combines the two. I was also able to make it so the `spec_version: 3` line is not necessary anymore, both styles can now be combined in the same spec.